tooling/noirc_abi/src/lib.rs

```diff
@@ -219,7 +222,7 @@ pub struct Abi {
     /// A map from the ABI's parameters to the indices they are written to in the [`WitnessMap`].
     /// This defines how to convert between the [`InputMap`] and [`WitnessMap`].
     pub param_witnesses: BTreeMap<String, Vec<Range<Witness>>>,
-    pub return_type: Option<AbiType>,
+    pub return_type: Option<(AbiType, AbiVisibility)>,
```

Comment:
This also needs to be reflected.

Reply:
Ok, the issue with this is that this type gets represented in JS/JSON as `[AbiType, Visibility]` whereas in `noirc_abi_wasm` you've said that the type is `{abi_type: AbiType, visibility: Visibility}` so these don't match up.

We could change `noirc_abi_wasm` to look for a `[AbiType, Visibility]` but when you pass in a JSON value, TS can't tell that the `return_type` field is trying to be a tuple and so it can only tell that it is the more general `(AbiType | Visibility)[]`. There's not really a way around this while having a tuple without requiring type assertions as we need to tell TS that we're passing in a `[AbiType, Visibility]` specifically.

Alternatively, we can make an actual `struct SomeReturnishType {type: AbiType, visibility: AbiVisibility}` type in the Rust code which will make TS type `{abi_type: AbiType, visibility: Visibility}` correct and the issue goes away.

So general solutions are:
1. Change the TS type to `[AbiType, Visibility]` and cast through `unknown` when you run into issues.
    a. We can follow up in future with a `parseAbi` function which will do type narrowing nicely.
2. Keep the TS type as is and update the rust code to match.

Reply:
I took the second option, it's a bit better on rust side (we use properties instead of .0, .1) and I am not sure how to do 1. on the js side. But I have no strong opinion here.